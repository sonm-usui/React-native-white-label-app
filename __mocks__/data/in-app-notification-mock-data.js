// @flow

import { MESSAGE_TYPE } from '../../src/api/api-constants'

export const inAppNotificationMockData = {
  notification: {
    senderName: 'Anonymous',
    senderImage: 'https://localhost/image.png',
    senderDID: 'senderDID1',
    identifier: 'receiverPairwiseDID1',
    title: 'You received a PROOF REQUEST',
    text: 'Test wants you to share Address',
    messageType: MESSAGE_TYPE.PROOF_REQUEST,
    messageId: 'messageId1',
  },
}

export const inAppNotificationEmptyMockData = {
  notification: null,
}
