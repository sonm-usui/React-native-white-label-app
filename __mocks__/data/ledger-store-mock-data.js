// @flow

import moment from 'moment'
import { STORE_STATUS } from '../../src/common/type-common'

export const transferFees = {
  transfer: '0.00001',
  refreshTime: moment().format(),
}

export const ledgerStoreWithTransferFees = {
  fees: {
    data: transferFees,
    status: STORE_STATUS.SUCCESS,
    error: null,
  },
}
