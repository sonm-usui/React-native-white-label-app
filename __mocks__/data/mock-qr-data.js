// @flow

export const mockAriesV1InvitationUrl =
  'http://localhost:80/?c_i=somebase64encodedstring'

export const mockAriesV1QrCode = {
  '@id': '7bf4f041-5ed4-4246-b309-396aeae830ab',
  '@type': 'did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/connections/1.0/invitation',
  label: 'alice',
  recipientKeys: ['6B5wQFpWbwJYB7qQYWydSwvCNKGoTZq6KvUe8TWBgPFY'],
  routingKeys: [
    '7re7C6zmg3p1r1hzHZSQJgt27wbycLeuzgopq1xNZqs6',
    '7G3LhXFKXKTMv7XGx1Qc9wqkMbwcU2iLBHL8x1JXWWC2',
  ],
  serviceEndpoint: 'https://agency.com/agency/msg',
}

export const mockAriesV1QrCodeNoRecipientKeys = {
  '@id': '7bf4f041-5ed4-4246-b309-396aeae830ab',
  '@type': 'did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/connections/1.0/invitation',
  label: 'alice',
  routingKeys: [
    '7re7C6zmg3p1r1hzHZSQJgt27wbycLeuzgopq1xNZqs6',
    '7G3LhXFKXKTMv7XGx1Qc9wqkMbwcU2iLBHL8x1JXWWC2',
  ],
  serviceEndpoint: 'https://agency.com/agency/msg',
}

export const mockAriesV1QrCodeNoRoutingKeys = {
  '@id': '7bf4f041-5ed4-4246-b309-396aeae830ab',
  '@type': 'did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/connections/1.0/invitation',
  label: 'alice',
  recipientKeys: ['6B5wQFpWbwJYB7qQYWydSwvCNKGoTZq6KvUe8TWBgPFY'],
  serviceEndpoint: 'https://agency.com/agency/msg',
}

export const mockAriesV1QrCodeNullRoutingKeys = {
  '@id': '7bf4f041-5ed4-4246-b309-396aeae830ab',
  '@type': 'did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/connections/1.0/invitation',
  label: 'alice',
  recipientKeys: ['6B5wQFpWbwJYB7qQYWydSwvCNKGoTZq6KvUe8TWBgPFY'],
  routingKeys: null,
  serviceEndpoint: 'https://agency.com/agency/msg',
}

export const mockAriesV1QrCodeNoLabel = {
  '@id': '7bf4f041-5ed4-4246-b309-396aeae830ab',
  '@type': 'did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/connections/1.0/invitation',
  recipientKeys: ['6B5wQFpWbwJYB7qQYWydSwvCNKGoTZq6KvUe8TWBgPFY'],
  routingKeys: null,
  serviceEndpoint: 'https://agency.com/agency/msg',
}

export const mockAriesV1QrCodeNullLabel = {
  '@id': '7bf4f041-5ed4-4246-b309-396aeae830ab',
  '@type': 'did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/connections/1.0/invitation',
  label: null,
  recipientKeys: ['6B5wQFpWbwJYB7qQYWydSwvCNKGoTZq6KvUe8TWBgPFY'],
  routingKeys: null,
  serviceEndpoint: 'https://agency.com/agency/msg',
}

export const mockEphemeralProofRequestQrCode = {
  '@id': '7bf4f041-5ed4-4246-b309-396aeae830bc',
  '@type': 'did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/connections/1.0/invitation',
  'request_presentations~attach': [
    {
      '@id': '7bf4f041-5ed4-4246-b309-396aeae830cd',
      'mime-type': 'application/json',
      data: {
        base64: 'oihkadnkfnakfadfda=/',
      },
    },
  ],
  comment: 'Sender Name',
  '~service': {
    recipientKeys: ['6B5wQFpWbwJYB7qQYWydSwvCNKGoTZq6KvUe8TWBgPFY'],
    routingKeys: [
      '7re7C6zmg3p1r1hzHZSQJgt27wbycLeuzgopq1xNZqs6',
      '7G3LhXFKXKTMv7XGx1Qc9wqkMbwcU2iLBHL8x1JXWWC2',
    ],
    serviceEndpoint: 'https://agency.com/agency/msg',
  },
}
