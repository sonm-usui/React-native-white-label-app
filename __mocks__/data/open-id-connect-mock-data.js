// @flow

import { OPEN_ID_CONNECT_STATE } from '../../src/open-id-connect/open-id-connect-actions'

export const mockOpenIdConnectRequest1Id =
  'a35dc33c-b88f-4c55-a4b1-1a9451b54cf7'

export const mockOpenIdConnectRequestData1 = {
  id: mockOpenIdConnectRequest1Id,
  jwtAuthenticationRequest: {
    body: {
      client_id: 'http://192.168.1.4:4005/cb',
      iss: 'did:sov:3C4TKPKo3wRi3AMmXVjaHc',
      nonce: 'GKSYeFnP2bSPBMvX8CFlg99lA_u_HQwC3xtbULY17ew',
      registration: {
        id_token_signed_response_alg: ['Ed25519'],
        jwks_uri: null,
        request_object_signing_alg: 'none',
      },
      response_mode: 'form_post',
      response_type: 'id_token',
      scope: 'openid did_authn',
      state: 'brYg8-5-1nlCcRtSAa5Y6n_W9WcpyMhdXerhGUNtAxI',
    },
    encodedSignature:
      'BukDc87nGbYnhdOZmcxTWTMZRQuuP8h3z2n0eqIWU5MzM-XT7z5QINgdw6Gl_lrLr_LUYiwyATmPe-jJKocSCQ==',
    header: {
      alg: 'Ed25519',
      kid:
        'did:sov:3C4TKPKo3wRi3AMmXVjaHc#2CFbxj7zy4HDKQkVKfPwH2LTKCoFRHoiCW7CbJxFwYj9',
      typ: 'JWT',
    },
  },
  oidcAuthenticationQrCode: {
    clientId: 'http://192.168.1.4:4005/cb',
    requestUri: 'http://192.168.1.4:4005/nRTsiNc_kwPane04tbKJRA',
    responseType: 'id_token',
    type: 'OIDC',
    version: '0.1',
  },
}

export const mockOpenIdConnectRequest1 = {
  error: undefined,
  oidcAuthenticationRequest: mockOpenIdConnectRequestData1,
  state: OPEN_ID_CONNECT_STATE.REQUEST_RECEIVED,
  version: 1,
}

export const mockOpenIdConnectReceivedState = {
  data: {
    [mockOpenIdConnectRequest1Id]: mockOpenIdConnectRequest1,
  },
  version: 1,
}
