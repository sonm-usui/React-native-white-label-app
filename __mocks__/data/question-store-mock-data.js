// @flow

import {
  COMMITEDANSWER_QUESTION_PROTOCOL,
  QUESTION_STATUS,
} from '../../src/question/type-question'
import { STORAGE_STATUS } from '../../src/common/type-common'

export const mockExternalLink = {
  text: 'An external link',
  src: 'https://somexternal.com/link',
}

export const mockExternalLinkNoText = {
  src: 'https://secondlink.com/external',
}

export const mockExternalLinks = [mockExternalLink, mockExternalLinkNoText]

export const mockQuestionPayload = {
  messageTitle: 'Message Title',
  messageText: 'Message Text',
  '@type': COMMITEDANSWER_QUESTION_PROTOCOL,
  messageId: 'MSG-ID',
  question_text: 'Question Text',
  question_detail: 'Question Detail',
  valid_responses: [
    { text: 'Answer 1', nonce: 'YES' },
    { text: 'Answer 2', nonce: 'No' },
  ],
  timing: { expires_time: '2017-06-01:05:07:07' },
  issuer_did: 'issuerDid',
  remoteDid: 'remoteDid',
  uid: '001',
  from_did: 'senderDID1',
  forDID: 'myPairwiseDid',
  externalLinks: [],
}

export const mockQuestionPayload2 = {
  messageTitle: 'Survey request for family members',
  messageText: 'Hi User,',
  '@type': COMMITEDANSWER_QUESTION_PROTOCOL,
  messageId: 'MSG-ID-2',
  question_text: 'How many members are in your family',
  question_detail:
    'Some more details regarding question that should scrolled along with some data',
  valid_responses: [
    { text: '0 to 2 people', nonce: 'akdfakjfkjdaadkgak' },
    { text: '3 to 5 people', nonce: 'hyjjjjuifhjkiutyojoh' },
    { text: '6 to 10 people', nonce: 'hyjjjjuifhjkiutdsyojoh' },
  ],
  timing: { expires_time: '2017-06-01:05:07:07' },
  issuer_did: 'issuerDid',
  remoteDid: 'remoteDid',
  uid: '002',
  from_did: 'senderDID1',
  forDID: 'myPairwiseDid',
  externalLinks: mockExternalLinks,
}

export const mockQuestionPayload3 = {
  messageTitle: 'Are you trying to sign in?',
  messageText: 'Hi User,',
  '@type': COMMITEDANSWER_QUESTION_PROTOCOL,
  messageId: 'MSG-ID-3',
  question_text: 'Are you trying to sign in to gmail.com?',
  question_detail: 'Are you trying to sign in to gmail.com?',
  valid_responses: [{ text: 'Yes', nonce: 'akdfakjfkjdaadkgak' }],
  timing: { expires_time: '2017-06-01:05:07:07' },
  issuer_did: 'issuerDid',
  remoteDid: 'remoteDid',
  uid: '003',
  from_did: 'senderDID1',
  forDID: 'myPairwiseDid',
  externalLinks: [],
}

export const mockQuestionReceivedState = {
  data: {
    [mockQuestionPayload.uid]: {
      payload: mockQuestionPayload,
      status: QUESTION_STATUS.RECEIVED,
      answer: null,
      answerMsgId: null,
      error: null,
    },
    [mockQuestionPayload2.uid]: {
      payload: mockQuestionPayload2,
      status: QUESTION_STATUS.RECEIVED,
      answer: null,
      answerMsgId: null,
      error: null,
    },
  },
  storageStatus: STORAGE_STATUS.RESTORE_SUCCESS,
}
