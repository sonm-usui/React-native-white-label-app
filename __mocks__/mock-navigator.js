// @flow
import * as React from 'react'
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'

jest.mock('react-native-screens', () => {
  const RealComponent = jest.requireActual('react-native-screens')
  RealComponent.enableScreens = jest.fn()
  return RealComponent
})

const Stack = createStackNavigator()
export const MockedNavigator = ({
  component,
  params = {},
}: {
  component: React.ElementType,
  params?: Object,
}) => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name="MockedScreen" initialParams={params}>
          {component}
        </Stack.Screen>
      </Stack.Navigator>
    </NavigationContainer>
  )
}
