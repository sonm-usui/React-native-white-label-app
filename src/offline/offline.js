// @flow

import React, { useEffect, useState, useMemo } from 'react'
import { View, StyleSheet, ActivityIndicator } from 'react-native'
import { connect, useSelector } from 'react-redux'
import { bindActionCreators } from 'redux'
import NetInfo, { useNetInfo } from '@react-native-community/netinfo'

import type { OfflineProps } from './type-offline'

import { color } from '../common/styles/constant'
import { offline } from './offline-store'
import { getUnacknowledgedMessages } from '../store/config-store'
import {
  getOfflineStatus,
  getIsLoading,
  getIsVcxPoolInitFailed,
  getIsVcxInitFailed,
  getIsGetMessagesFailed,
} from '../store/store-selector'
import { vcxInitPoolStart, vcxInitStart } from '../store/route-store'
import { CustomText, CustomView } from '../components'

export const Offline = ({
  offline,
  vcxInitPoolStart,
  vcxInitStart,
  getUnacknowledgedMessages,
  overlay,
}: OfflineProps) => {
  const netInfo = useNetInfo()
  const [isLoading, setIsLoading] = useState(false)

  const isOffline = useSelector(getOfflineStatus)
  const isPoolError = useSelector(getIsVcxPoolInitFailed)
  const isVcxError = useSelector(getIsVcxInitFailed)
  const isGetMessagesError = useSelector(getIsGetMessagesFailed)
  const isVcxLoading = useSelector(getIsLoading)

  const isShowLoader = useMemo(() => {
    return isOffline ? isLoading : isVcxLoading
  }, [isOffline, isLoading, isVcxLoading])

  const isNeedToReconnect = useMemo(() => {
    return isOffline || isVcxError || isPoolError || isGetMessagesError
  }, [isOffline, isVcxError, isPoolError, isGetMessagesError])

  const currentErrorForMessage = useMemo(() => {
    if (isOffline) {
      return 'internet'
    }
    if (isGetMessagesError) {
      return 'agent'
    }
    if (isPoolError) {
      return 'pool'
    }
    return ''
  }, [isOffline, isVcxError, isPoolError, isGetMessagesError])

  const [ignoreNetInfo, setIgnoreNetInfo] = useState(true)

  useEffect(() => {
    // FIXME: netInfo always return false as the first event even if devices are connected to internet
    // Workaround: ignore null and ignore false only the first time
    if (typeof netInfo.isInternetReachable === 'boolean') {
      if (ignoreNetInfo) {
        setIgnoreNetInfo(false)
      } else {
        offline(!netInfo.isInternetReachable)
      }
    }
  }, [netInfo])

  const internetReconnect = () => {
    setIsLoading(true)
    const req = setInterval(async () => {
      const state = await NetInfo.fetch()
      offline(!state.isConnected)
    }, 500)
    setTimeout(() => {
      setIsLoading(false)
      clearInterval(req)
    }, 3000)
  }

  const getMessageRetry = () => getUnacknowledgedMessages()

  const poolReconnect = () => vcxInitPoolStart()

  const vcxReconnect = () => vcxInitStart()

  const initReconnectFunction = () => {
    if (isOffline) {
      return internetReconnect()
    }
    if (isVcxError) {
      return vcxReconnect()
    }
    if (isGetMessagesError) {
      return getMessageRetry()
    }
    if (isPoolError) {
      return poolReconnect()
    }
  }

  return isNeedToReconnect && overlay ? (
    <CustomView doubleVerticalSpace horizontalSpace bg={'yellow'}>
      {!isShowLoader ? (
        <CustomText bg center>
          {`No ${currentErrorForMessage} connection detected. `}
          <CustomText bg bold underline onPress={initReconnectFunction}>
            Reconnect
          </CustomText>
        </CustomText>
      ) : (
        <View style={styles.overlay}>
          <CustomText bg center>
            Attempting to reconnect...{' '}
          </CustomText>
          <ActivityIndicator size="large" color={color.actions.fifth} />
        </View>
      )}
    </CustomView>
  ) : null
}

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      offline,
      vcxInitPoolStart,
      vcxInitStart,
      getUnacknowledgedMessages,
    },
    dispatch
  )

export default connect(null, mapDispatchToProps)(Offline)

const styles = StyleSheet.create({
  overlay: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
})
