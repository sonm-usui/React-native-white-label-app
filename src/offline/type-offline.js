// @flow
import { vcxInitPoolStart, vcxInitStart } from '../store/route-store'
import { getUnacknowledgedMessages } from '../store/config-store'

export const OFFLINE_STATUS = 'OFFLINE_STATUS'

export type OfflineStore = {
  offline: boolean,
}

export type OfflineActions = OfflineStatus

export type OfflineStatus = {
  type: typeof OFFLINE_STATUS,
  offline: boolean,
}

export type OfflineProps = {
  overlay?: boolean,
  offline: (offline: boolean) => void,
  render?: (isConnected: ?boolean) => any,
  vcxInitPoolStart: typeof vcxInitPoolStart,
  vcxInitStart: typeof vcxInitStart,
  getUnacknowledgedMessages: typeof getUnacknowledgedMessages,
}
