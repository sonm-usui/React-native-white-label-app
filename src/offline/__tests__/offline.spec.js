// @flow
import React from 'react'
import 'react-native'
import { Text } from 'react-native'
import renderer from 'react-test-renderer'
import merge from 'lodash.merge'
import { Provider } from 'react-redux'

import { getStore } from '../../../__mocks__/static-data'
import { configStoreNotHydratedInstalledVcxInit } from '../../../__mocks__/data/config-store-mock-data'
import { Offline } from '../offline'

const getProps = () => {
  return {
    offline: jest.fn(),
    overlay: true,
    vcxInitPoolStart: jest.fn(),
    vcxInitStart: jest.fn(),
    getUnacknowledgedMessages: jest.fn(),
  }
}

const setup = (currentStore) => {
  const props = getProps()
  const component = renderer.create(
    <Provider store={currentStore}>
      <Offline {...props}></Offline>
    </Provider>
  )
  const instance = component.getInstance()

  return { props, component, instance }
}

describe('<Offline /> for offline case', () => {
  const currentState = getStore().getState()
  const store = {
    ...getStore(),
    getState() {
      return merge(
        {},
        {
          ...currentState,
          offline: {
            offline: true,
          },
        }
      )
    },
  }
  it('should match snapshot for offline case', () => {
    const { component } = setup(store)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('current text error for offline case', () => {
    const { component } = setup(store)
    let tree = component.root.findByType(Text).props.children
    expect(tree[0] + tree[1].props.children).toBe(
      'No internet connection detected. Reconnect'
    )
  })

  it('click is working for offline case', () => {
    const { component } = setup(store)
    expect(component.root.findAllByType(Text).length).toBe(2)
    let tree = component.root.findByType(Text).props.children
    tree[1].props.onPress()
    expect(component.root.findAllByType(Text).length).toBe(1)
  })
})

describe('<Offline /> for disconnect agent case', () => {
  const currentState = getStore().getState()
  const store = {
    ...getStore(),
    getState() {
      return merge(
        {},
        {
          ...currentState,
          config: {
            ...configStoreNotHydratedInstalledVcxInit,
            isVcxInitFailed: true,
          },
        }
      )
    },
  }
  it('should match snapshot for disconnect agent case', () => {
    const { component } = setup(store)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('click is working for disconnect agent case', () => {
    const { component } = setup(store)
    let tree = component.root.findByType(Text).props.children
    tree[1].props.onPress()
  })
})

describe('<Offline /> for fail get messages case', () => {
  const currentState = getStore().getState()
  const store = {
    ...getStore(),
    getState() {
      return merge(
        {},
        {
          ...currentState,
          config: {
            ...configStoreNotHydratedInstalledVcxInit,
            isGetMessagesFailed: true,
          },
        }
      )
    },
  }
  it('should match snapshot for fail get messages case', () => {
    const { component } = setup(store)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('current text error for fail get messages case', () => {
    const { component } = setup(store)
    let tree = component.root.findByType(Text).props.children
    expect(tree[0] + tree[1].props.children).toBe(
      'No agent connection detected. Reconnect'
    )
  })

  it('click is working for fail get messages case', () => {
    const { component } = setup(store)
    let tree = component.root.findByType(Text).props.children
    tree[1].props.onPress()
  })
})

describe('<Offline /> for disconnect pool case', () => {
  const currentState = getStore().getState()
  const store = {
    ...getStore(),
    getState() {
      return merge(
        {},
        {
          ...currentState,
          config: {
            ...configStoreNotHydratedInstalledVcxInit,
            isVcxPoolInitFailed: true,
          },
        }
      )
    },
  }
  it('should match snapshot for disconnect pool case', () => {
    const { component } = setup(store)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('current text error for disconnect pool case', () => {
    const { component } = setup(store)
    let tree = component.root.findByType(Text).props.children
    expect(tree[0] + tree[1].props.children).toBe(
      'No pool connection detected. Reconnect'
    )
  })

  it('click is working for disconnect pool case', () => {
    const { component } = setup(store)
    let tree = component.root.findByType(Text).props.children
    tree[1].props.onPress()
  })
})
