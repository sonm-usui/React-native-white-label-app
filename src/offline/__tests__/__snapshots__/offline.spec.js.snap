// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Offline /> for disconnect agent case should match snapshot for disconnect agent case 1`] = `
<View
  bg="yellow"
  doubleVerticalSpace={true}
  horizontalSpace={true}
  style={
    Object {
      "backgroundColor": "#EB9B2D",
      "paddingHorizontal": 15,
      "paddingVertical": 16,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "fontSize": 17,
        },
        Object {
          "fontFamily": "System",
        },
        Object {},
        null,
        Object {
          "textAlign": "center",
        },
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ]
    }
    suppressHighlighting={true}
  >
    No  connection detected. 
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "fontSize": 17,
          },
          Object {
            "fontFamily": "System",
          },
          Object {},
          Object {
            "fontWeight": "bold",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          Object {
            "textDecorationLine": "underline",
          },
          null,
        ]
      }
      suppressHighlighting={true}
    >
      Reconnect
    </Text>
  </Text>
</View>
`;

exports[`<Offline /> for disconnect pool case should match snapshot for disconnect pool case 1`] = `
<View
  bg="yellow"
  doubleVerticalSpace={true}
  horizontalSpace={true}
  style={
    Object {
      "backgroundColor": "#EB9B2D",
      "paddingHorizontal": 15,
      "paddingVertical": 16,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "fontSize": 17,
        },
        Object {
          "fontFamily": "System",
        },
        Object {},
        null,
        Object {
          "textAlign": "center",
        },
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ]
    }
    suppressHighlighting={true}
  >
    No pool connection detected. 
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "fontSize": 17,
          },
          Object {
            "fontFamily": "System",
          },
          Object {},
          Object {
            "fontWeight": "bold",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          Object {
            "textDecorationLine": "underline",
          },
          null,
        ]
      }
      suppressHighlighting={true}
    >
      Reconnect
    </Text>
  </Text>
</View>
`;

exports[`<Offline /> for fail get messages case should match snapshot for fail get messages case 1`] = `
<View
  bg="yellow"
  doubleVerticalSpace={true}
  horizontalSpace={true}
  style={
    Object {
      "backgroundColor": "#EB9B2D",
      "paddingHorizontal": 15,
      "paddingVertical": 16,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "fontSize": 17,
        },
        Object {
          "fontFamily": "System",
        },
        Object {},
        null,
        Object {
          "textAlign": "center",
        },
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ]
    }
    suppressHighlighting={true}
  >
    No agent connection detected. 
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "fontSize": 17,
          },
          Object {
            "fontFamily": "System",
          },
          Object {},
          Object {
            "fontWeight": "bold",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          Object {
            "textDecorationLine": "underline",
          },
          null,
        ]
      }
      suppressHighlighting={true}
    >
      Reconnect
    </Text>
  </Text>
</View>
`;

exports[`<Offline /> for offline case should match snapshot for offline case 1`] = `
<View
  bg="yellow"
  doubleVerticalSpace={true}
  horizontalSpace={true}
  style={
    Object {
      "backgroundColor": "#EB9B2D",
      "paddingHorizontal": 15,
      "paddingVertical": 16,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "fontSize": 17,
        },
        Object {
          "fontFamily": "System",
        },
        Object {},
        null,
        Object {
          "textAlign": "center",
        },
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ]
    }
    suppressHighlighting={true}
  >
    No internet connection detected. 
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "fontSize": 17,
          },
          Object {
            "fontFamily": "System",
          },
          Object {},
          Object {
            "fontWeight": "bold",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          Object {
            "textDecorationLine": "underline",
          },
          null,
        ]
      }
      suppressHighlighting={true}
    >
      Reconnect
    </Text>
  </Text>
</View>
`;
